### PR TITLE
Storageaccount tlsversion

### PR DESCRIPTION
**Pull request by paultaiton**
##### SUMMARY

This change adds support to azure_rm_storageaccounts.py for the "MinimumTlsVersion" parameter for new and updated storage accounts.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_storageaccount
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by paultaiton** [comment]
This is the first code of any significance I've written with Azure python SDK and for Ansible modules. I made a lot of assumptions to get this working, so I really appreciate any help with reviewing and getting this to an acceptable quality.

**Comment by Fred-sun** [comment]
@paultaiton Thank you for your contribution, we will review and promote PR merger as soon as possible. Can you add test cases to this feature? Thank you!
```e.g.
tests/integration/targets/azure_rm_storageaccount/tasks/main.yml

**Comment by paultaiton** [comment]
@Fred-sun 
I have not done ansible tests before, but I'll see what I can throw together and will let you know when I think I have it.
Thanks Fred.

**Comment by paultaiton** [comment]
@Fred-sun 
I added some tests, but in the process realized that I did not add sufficient support for the _info module. That has also been added in the latest commit.

During my own testing of adding the TLS feature, I realized that there were errors with the code as it existed, and had to make modifications outside the scope of MinimumTlsVersion to make execution complete without errors.

As mentioned I added some code for the integration tests, but I have no idea how to run it. I THINK it is correct syntactically and logically, but lacking the actual test I cannot confirm. Could I please ask that you either run the test for me, or point me in the right direction for documentation on how to accomplish that myself?

Thanks again for your help.

**Comment by Fred-sun** [comment]
> @Fred-sun
> I added some tests, but in the process realized that I did not add sufficient support for the _info module. That has also been added in the latest commit.
> 
> During my own testing of adding the TLS feature, I realized that there were errors with the code as it existed, and had to make modifications outside the scope of MinimumTlsVersion to make execution complete without errors.
> 
> As mentioned I added some code for the integration tests, but I have no idea how to run it. I THINK it is correct syntactically and logically, but lacking the actual test I cannot confirm. Could I please ask that you either run the test for me, or point me in the right direction for documentation on how to accomplish that myself?
> 
> Thanks again for your help.

I will check your change! Thank you very much!

File: plugins/modules/azure_rm_storageaccount.py [hunk 1]
```diff
@@ -680,6 +694,18 @@ def update_account(self):
                 except Exception as exc:
                     self.fail("Failed to update account type: {0}".format(str(exc)))
 
+        if bool(self.minimum_tls_version) != self.account_dict.get('minimum_tls_version'):
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]:
```suggestion
        if self.minimum_tls_version is not None and self.minimum_tls_version != self.account_dict.get('minimum_tls_version'):
```

File: plugins/modules/azure_rm_storageaccount_info.py [hunk 2]
```diff
@@ -517,18 +517,19 @@ def account_obj_to_dict(self, account_obj, blob_service_props=None):
             id=account_obj.id,
             name=account_obj.name,
             location=account_obj.location,
-            access_tier=(account_obj.access_tier.value
+            access_tier=(account_obj.access_tier
                          if account_obj.access_tier is not None else None),
-            account_type=account_obj.sku.name.value,
-            kind=account_obj.kind.value if account_obj.kind else None,
-            provisioning_state=account_obj.provisioning_state.value,
+            account_type=account_obj.sku.name, #.value,
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount_info.py]:
```suggestion
            account_type=account_obj.sku.name,
```

**Review by Fred-sun — COMMENTED** [review]
@paultaiton Please help change as suggested below. Thank you!

File: plugins/modules/azure_rm_storageaccount.py [hunk 3]
```diff
@@ -94,6 +94,17 @@
...
+    minimum_tls_version:
+        description:
+            - The minimum required version of Transport Layer Security (TLS) for requests to a storage account.
+        default: 'TLS1_0'
+        choices:
+            - TLS1_0
+            - TLS1_1
+            - TLS1_2
+        version_added: "2.10"
+        version_added_collection: "azure.azcollection v0.4.0"
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]:
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
@paultaiton Thanks!

**Comment by paultaiton** [comment]
Suggestions look good.

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by paultaiton** [comment]
Hello @Fred-sun .

Was there any further action required on my part?  I'm not sure what the next steps are.

**Comment by Fred-sun** [comment]
> Hello @Fred-sun .
> 
> Was there any further action required on my part? I'm not sure what the next steps are.

Because the maintainer was busy with other things during this time, there are no time to review these PRs. I will push maintainer to handle these PRs today or tomorrow. If it causes you inconvenience, please forgive me. Thank you!

**Comment by paultaiton** [comment]
No problem, just wanted to make sure it was not waiting on me. Appreciate the response.

**Comment by paultaiton** [comment]
Thanks for your help @haiyuazhang , and especially for your help @Fred-sun . It is much appreciated.

**Pull request merged**